### PR TITLE
CESM Galaxy tool

### PR DESCRIPTION
**Pull request by annefou**
### Community Earth System Model

- [x] Support main scientifically tested configurations (compsets/grids)
- [x] Support user modifications 
- [x] Support user namelist changes
- [x] Allow user to customize runs (duration, etc.)
- [x] Add a test to run a simple default configuration
- [x] Add a test to run a customized simulation

### CODE REVIEW

File: tools/cesm/cesm.xml [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+            </param>
+            <when value="FKESSLER">
+                <param name="resolution" type="select" label="Model resolution">
+                    <option value="--res f09_f09_mg17">f09_f09_mg17</option>
+                    <option value="--res f19_f19_mg17">f19_f19_mg17</option>
+                </param>
+            </when>
+            <when value="F2000climo">
+                <param name="resolution" type="select" label="Model resolution">
+                    <option value="--res f09_f09_mg17">f09_f09_mg17</option>
```
Comment (j34ni) [inline tools/cesm/cesm.xml:101]: add
<option value="--res f19_f19_mg17 --run-unsupported">f19_f19_mg17</option>

File: tools/cesm/test-data/case_info.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+cat: 'work/FKESSLER/run/rof.log.*': No such file or directory
```
Comment (bgruening) [inline tools/cesm/test-data/case_info.txt:1]: that does not look good :)

File: tools/cesm/cesm.xml [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+    <requirements>
+        <requirement type="package" version="2.1.3">cesm</requirement>
+        <requirement type="package" version="1.32">tar</requirement>
+        <requirement type="package" version="2.35">binutils</requirement>
+        <requirement type="package" version="3">python</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        #set galaxy_slots = os.getenv("GALAXY_SLOTS", 4)
+        export HOME=`pwd` &&
+        export USER='jovyan' &&
```
Comment (bgruening) [inline tools/cesm/cesm.xml:16]: why is this needed?

File: tools/cesm/test-data/case_info.txt (hunk 2 above)
Comment (annefou) [reply tools/cesm/test-data/case_info.txt:1]: Indeed! the test configuration does not use all the components (no runoff model active). I think I changed this part but the test check was too loose.

File: tools/cesm/cesm.xml (hunk 3 above)
Comment (annefou) [reply tools/cesm/cesm.xml]: - I always have problems with HOME. Most climate tools need to have configuration files in the HOME; so I re-define it.
- For USER, I am not sure. I need it when I use the corresponding "bio-container". Maybe I could use a more relevant username?
- For GALAXY_SLOTS: I have no experience in using parallel jobs in Galaxy. Our tool is usually run on HPC with SLURM and we specify both the number of tasks per node and number of nodes. With Galaxy, I was not sure what to put and I don't know how it can/will work. I used GALAXY_SLOTS as the number of tasks per node and use node=1.

Comment (bgruening) [reply tools/cesm/cesm.xml]: I think HOME is not set with a recent profile= version.

> For USER, I am not sure. I need it when I use the corresponding "bio-container". Maybe I could use a more relevant username?

Mh, jovyan is the user from the jupyter container, but no other container I think.

> GALAXY_SLOTS

do this here:

`bash $__tool_directory__/create_config_machines.sh \${GALAXY_SLOTS:-4}`

Comment (annefou) [reply tools/cesm/cesm.xml]: Thanks!

`USER` is not defined in the container but our tool fails if this environment variable is not defined.

Comment (bgruening) [reply tools/cesm/cesm.xml]: oh I see ... maybe a more meaningful name, like "USER='FAKEUSER'

Comment (annefou) [reply tools/cesm/cesm.xml]: `bash $__tool_directory__/create_config_machines.sh \${GALAXY_SLOTS:-4}` does not seem to work. Maybe this is related to planemo? 
Using
`#set galaxy_slots = os.getenv("GALAXY_SLOTS", 4)
bash $__tool_directory__/create_config_machines.sh $galaxy_slots &&
`
seems to work. This is really weird!

Comment (bgruening) [reply tools/cesm/cesm.xml]: what means does not work? How do you test this?

Comment (annefou) [reply tools/cesm/cesm.xml]: You are right.  I probably do not know how to test it properly. Here is what I do:

```
export GALAXY_SLOTS=8
planemo t tools/cesm --docker 
```

I also tried:

```
GALAXY_SLOTS=8 planemo t tools/cesm --docker 
```

And when the tool runs, it sets GALAXY_SLOTS to 1

Comment (bgruening) [reply tools/cesm/cesm.xml]: its always one, as this is default set by Galaxy. We use this pattern everywhere in tools-iuc, so it would be surprising if this does not work. Indeed you need to set this in a job_conf file in Galaxy. But this is Admin stuff that I will do for you.

I'm confused by the python trick should work and the bash thing not. But lets stick to the bash stuff and do a release. We can see how it behaves on usegalaxy.eu

File: tools/cesm/cesm.xml [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+        <requirement type="package" version="2.1.3">cesm</requirement>
+        <requirement type="package" version="1.32">tar</requirement>
+        <requirement type="package" version="2.35">binutils</requirement>
+        <requirement type="package" version="3">python</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        #set galaxy_slots = os.getenv("GALAXY_SLOTS", 4)
+        export HOME=`pwd` &&
+        export USER='FAKEUSER' &&
+        export AR=x86_64-conda_cos6-linux-gnu-ar &&
```
Comment (bgruening) [inline tools/cesm/cesm.xml:16]: This is very conda specific. It is ok, I just wanted to flag you that. It probably only works with conda and biocontaintainers.

File: tools/cesm/cesm.xml [hunk 5]
```diff
@@ -0,0 +1,465 @@
...
+            #for $s in $condi_user_mods.usermods
+                echo "User mod ${s.srcmod} ${s.srcmod.element_identifier}" &&
+                ln -s '${s.srcmod}' 'usermods_dirs/SourceMods/${s.srcmod.element_identifier}' &&
+            #end for
+        #end if
+        echo "create case" &&
+        create_newcase --case '$casename' --compset '$condi_compset.compset' --machine galaxy 
+            --pesfile '$__tool_directory__/config_pes.xml'
+        #if $condi_user_mods.add_changes
+            --user-mods-dir \$HOME/usermods_dirs
```
Comment (bgruening) [inline tools/cesm/cesm.xml:45]: why HOME here? `./usermods_dirs`?

File: tools/cesm/cesm.xml [hunk 6]
```diff
@@ -0,0 +1,465 @@
...
+        #end if
+            --res $condi_compset.resolution  > $case_info  2>&1 &&
```
Comment (bgruening) [inline tools/cesm/cesm.xml:47]: single-quote case_info

File: tools/cesm/cesm.xml [hunk 7]
```diff
@@ -0,0 +1,465 @@
...
+        #end if
+        echo "create case" &&
+        create_newcase --case '$casename' --compset '$condi_compset.compset' --machine galaxy 
+            --pesfile '$__tool_directory__/config_pes.xml'
+        #if $condi_user_mods.add_changes
+            --user-mods-dir \$HOME/usermods_dirs
+        #end if
+            --res $condi_compset.resolution  > $case_info  2>&1 && 
+        echo "create done" &&
+        cd '$casename' &&
```
Comment (bgruening) [inline tools/cesm/cesm.xml:49]: If you like you can also use popd and pushd for that (I hope that is also included in the containers :))

File: tools/cesm/cesm.xml [hunk 8]
```diff
@@ -0,0 +1,465 @@
...
+            cp '$adv_glc.user_nl_glc_customization' user_nl_cism &&
+        #end if
+        #if str($adv_ocn.user_nl_ocn_customization) != 'None'
+            cp '$adv_ocn.user_nl_ocn_customization' user_nl_ocn &&
+        #end if
+        #if str($adv_ice.user_nl_ice_customization) != 'None'
+            cp '$adv_ice.user_nl_ice_customization' user_nl_cice &&
+        #end if
+        #for $i in $adv_lnd.inputs
+            ln -s '${i.input}' ${i.input.element_identifier} &&
```
Comment (bgruening) [inline tools/cesm/cesm.xml:90]: Those files need to be next to each other?

File: tools/cesm/cesm.xml [hunk 9]
```diff
@@ -0,0 +1,465 @@
...
+           ( echo "Case failed" && cd .. && tar cf $work work && 
+             bash '$__tool_directory__/save_logs.sh' 'work/$casename/run/' logs_dir '$case_info' &&
+             exit 1 )
+        ) &&
+        echo "CURRENT PATH" &&
+        echo "\$HOME"       &&
+        cd "\$HOME"         && 
+        mkdir -p restart                              &&
+        #if (str($adv_restart.restart_option) != 'never')
+            bash '$__tool_directory__/save_restart.sh' 'work/$casename/run/' $casename restart '$case_info' &&
```
Comment (bgruening) [inline tools/cesm/cesm.xml:123]: casename probably also needs to be quoted?

File: tools/cesm/cesm.xml [hunk 10]
```diff
@@ -0,0 +1,465 @@
...
+            cat 'work/$casename/run/rpointer'.* > $rinfo 2>> '$case_info'  &&
+        #end if
+        cd restart &&
```
Comment (bgruening) [inline tools/cesm/cesm.xml:126]: popd and pushd

File: tools/cesm/cesm.xml [hunk 11]
```diff
@@ -0,0 +1,465 @@
...
+        tar cvf $restart . &&
+        cd "\$HOME"        && 
+        #if (str($adv_period.hist_option) != 'never')
+            cp 'work/$casename/run/$casename'.*.h*.*.nc output_dir/ 2>> '$case_info' &&
+        #end if
+        bash '$__tool_directory__/save_logs.sh' 'work/$casename/run/' logs_dir '$case_info' &&
+        #if $save_workdir
+            tar cf '$work' work  &&
+        #else
+            touch '$work' &&
```
Comment (bgruening) [inline tools/cesm/cesm.xml:136]: is that really needed?

File: tools/cesm/cesm.xml [hunk 12]
```diff
@@ -0,0 +1,465 @@
...
+        #end if
+        echo "MODEL RUN COMPLETED"
+    ]]></command>
+    <inputs>
+        <conditional name="input_type">
+            <param name="tarball" type="boolean" value="false" label="Provide inputdata as a tarball"/>
```
Comment (bgruening) [inline tools/cesm/cesm.xml:142]: According to IUC guidelines conditionals should alsways use select and not boolean

File: tools/cesm/cesm.xml [hunk 13]
```diff
@@ -0,0 +1,465 @@
...
+            <when value="true">
+                <param name="inputdata" type="data" format='tar' label="inputdata for running CESM (tarball)" />
+            </when>
+            <when value="false">
+                <param name="input_data_collection" format="netcdf" type="data_collection" collection_type="list" label="Select input data" help="Select all the netCDF files you would need for running the chosen compset and resolution"/>
```
Comment (bgruening) [inline tools/cesm/cesm.xml:147]: simple input but as multiple="true" instead of a data_collection

File: tools/cesm/cesm.xml [hunk 14]
```diff
@@ -0,0 +1,465 @@
...
+                <param name="mapping" type="data" format='txt' label="Text file for mapping files in inputdata collection to physical files"/>
```
Comment (bgruening) [inline tools/cesm/cesm.xml:148]: what does that mapping do? its not tested either is it?

Comment (bgruening) [reply tools/cesm/cesm.xml:148]: I mean the input_collection are physical files, no?

File: tools/cesm/cesm.xml [hunk 15]
```diff
@@ -0,0 +1,465 @@
...
+            </when>
+            <when value="BHIST">
+                <param name="resolution" type="select" label="Model resolution">
+                    <option value="f09_g17_gl4">f09_g17_gl4</option>
+                    <option value="f09_g17">f09_g17</option>
+                    <option value="f19_g17_gl4">f19_g17_gl4</option>
+                    <option value="f19_g17">f19_g17</option>
+                </param>
+            </when>
+	</conditional>
```
Comment (bgruening) [inline tools/cesm/cesm.xml:217]: indentation

File: tools/cesm/cesm.xml [hunk 16]
```diff
@@ -0,0 +1,465 @@
...
+            </repeat>
+        </section>
+        <section name="adv_ocn" title="Ocean namelist user customization" expanded="false">
+            <param name="user_nl_ocn_customization" type="data" format="txt" optional="true" label="OCN User namelist (ocn_in)"/>
+            <repeat name="inputs" title="Ocean Customized Input Files">
+               <param name="input" type="data" format="netcdf" label="Input for OCN component"/>
+            </repeat>
+        </section>
+        <conditional name="condi_user_mods">
+            <param name="add_changes" type="boolean" value="false" label="user-modified source code"/>
```
Comment (bgruening) [inline tools/cesm/cesm.xml:340]: select instead of boolean

File: tools/cesm/cesm.xml [hunk 17]
```diff
@@ -0,0 +1,465 @@
...
+        <section name="adv_ice" title="Sea-ice namelist user customization" expanded="false">
+            <param name="user_nl_ice_customization" type="data" format="txt" optional="true" label="ICE User namelist (ice_in)"/>
+            <repeat name="inputs" title="Sea-ice Customized Input Files">
+               <param name="input" type="data" format="netcdf" label="Input for ICE component"/>
+            </repeat>
+        </section>
+        <section name="adv_ocn" title="Ocean namelist user customization" expanded="false">
+            <param name="user_nl_ocn_customization" type="data" format="txt" optional="true" label="OCN User namelist (ocn_in)"/>
+            <repeat name="inputs" title="Ocean Customized Input Files">
+               <param name="input" type="data" format="netcdf" label="Input for OCN component"/>
```
Comment (bgruening) [inline tools/cesm/cesm.xml:336]: why not multiple=True instead of repeat?

File: tools/cesm/config_compilers.xml [hunk 18]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0"?>
+
+<config_compilers version="2.0">
+
+  <compiler COMPILER="gnu" MACH="galaxy">
+    <LD>mpifort</LD>
+    <SFC>x86_64-conda_cos6-linux-gnu-gfortran</SFC>
```
Comment (bgruening) [inline tools/cesm/config_compilers.xml:7]: AR can not be added here?

File: tools/cesm/config_machines.xml [hunk 19]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0"?>
+<config_machines>
+  <machine MACH="galaxy">
+    <DESC>
+          VM
+          x4 CPUs
+          Gold Centos 7
```
Comment (bgruening) [inline tools/cesm/config_machines.xml:7]: what implications does that have?

**Review by bgruening — COMMENTED** [review]
Hi Anne, I added a few comments inline.

File: tools/cesm/cesm.xml (hunk 3 above)
Comment (bgruening) [reply tools/cesm/cesm.xml]: Actually, we really __need__ to use \$GALAXY_SLOTS. The difference is that the cheetah variable gets set on the webserver, during job-preparation. But we want to have this set during job-scheduling. I changed that.

File: tools/cesm/cesm.xml (hunk 4 above)
Comment (annefou) [reply tools/cesm/cesm.xml:16]: True. The configuration files I created (`config_machines.xml` and `config_compilers.xml`) are meant to be used for conda and biocontainers. We have different configruation files when running on different platforms. I know it has been a problem for use; for instance when running on ARM processors. I am not sure how to make it better.

File: tools/cesm/cesm.xml (hunk 7 above)
Comment (annefou) [reply tools/cesm/cesm.xml:49]: ? I never used popd and pushd? Do you have an example?

File: tools/cesm/cesm.xml (hunk 4 above)
Comment (bgruening) [reply tools/cesm/cesm.xml:16]: Something to worry later about :)

File: tools/cesm/cesm.xml (hunk 8 above)
Comment (annefou) [reply tools/cesm/cesm.xml:90]: They need to have the name given by users (and not dataset... from galaxy) because the filename will be referenced by the user in the corresponding namelist.

File: tools/cesm/cesm.xml (hunk 7 above)
Comment (bgruening) [reply tools/cesm/cesm.xml:49]: https://en.wikipedia.org/wiki/Pushd_and_popd

its like cd, it just puts the dir on a stack and you can go back

pushd ./foo # in foo
pushd ./bar # in foo/bar
popd # in foo
popd  in ./

File: tools/cesm/cesm.xml (hunk 11 above)
Comment (annefou) [reply tools/cesm/cesm.xml:136]: I don't know. The thing is that I don't know how to make "optional" outputs. What will happen if I don't create $work?

File: tools/cesm/cesm.xml (hunk 13 above)
Comment (annefou) [reply tools/cesm/cesm.xml:147]: Ok. I have never done that! Thanks.

File: tools/cesm/cesm.xml [hunk 20]
```diff
@@ -0,0 +1,465 @@
...
+            <when value="true">
+                <repeat name="usermods" title="SourceMods files">
+                    <param name="srcmod" type="data" format="txt" label="Modified source file"/>
+                </repeat>
+            </when>
+            <when value="false"/>
+        </conditional>
+    </inputs>
+    <outputs>
+        <data name="work" format="tar" label="${tool.name} on ${casename} (workdir)"/>
```
Comment (bgruening) [inline tools/cesm/cesm.xml:350]: you can use `filter` to make this an optional output:

https://github.com/galaxyproject/tools-iuc/blob/84c175ed516a3b0074ca06d6546b76f1cbc4fb68/tools/umi_tools/umi-tools_extract.xml#L150

File: tools/cesm/cesm.xml (hunk 14 above)
Comment (annefou) [reply tools/cesm/cesm.xml:148]: I wanted to have a test but I do not know how to create a collection in tests...

Input_collection are physical files but the model does not expect to have a "flat" structures. Files need to be in specific folders. so this mapping file is meant to tell in which folder to put a specific file.

File: tools/cesm/cesm.xml (hunk 11 above)
Comment (bgruening) [reply tools/cesm/cesm.xml:136]: With `filter` https://github.com/NordicESMhub/galaxy-tools/pull/51/files#r650399364

File: tools/cesm/cesm.xml (hunk 17 above)
Comment (annefou) [reply tools/cesm/cesm.xml:336]: Because I did not know about it! Thanks!

File: tools/cesm/config_compilers.xml (hunk 18 above)
Comment (annefou) [reply tools/cesm/config_compilers.xml:7]: I would love to but `config_compilers.xml` has a very rigid format (it is read in Fortran by the model) and does not accept new keyword.

File: tools/cesm/config_machines.xml (hunk 19 above)
Comment (annefou) [reply tools/cesm/config_machines.xml:7]: None. It is a comment. I could put something different. Maybe something related to Galaxy?

File: tools/cesm/cesm.xml (hunk 14 above)
Comment (bgruening) [reply tools/cesm/cesm.xml:148]: Do you have an example file? How many folders? Could you not ask the users for X different collections for X different folders?

File: tools/cesm/config_machines.xml (hunk 19 above)
Comment (bgruening) [reply tools/cesm/config_machines.xml:7]: Ok, then we don't bother. Just wanted to check if we need to run it on CentOS 7 or so

File: tools/cesm/cesm.xml (hunk 7 above)
Comment (bgruening) [reply tools/cesm/cesm.xml:49]: Anne, not sure if useful, but if you change a lot the directories it might be easier. really not needed.

Comment (annefou) [reply tools/cesm/cesm.xml:49]: ok. But at least now I know! Thanks!

File: tools/cesm/cesm.xml (hunk 14 above)
Comment (annefou) [reply tools/cesm/cesm.xml:148]: It depends on the configuration chosen by the user. And there are folders, sub-folders and sub-sub-folders.
For one configuration:
```
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_bc_a4_anthro_surface_2000climo_0.9x1.25_c20170608.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_bc_a4_bb_surface_2000climo_0.9x1.25_c20170322.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_DMS_bb_surface_2000climo_0.9x1.25_c20170322.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_DMS_other_surface_2000climo_0.9x1.25_c20170322.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_num_a1_so4_contvolcano_vertical_2000climo_0.9x1.25_c20170724.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_num_a2_so4_contvolcano_vertical_2000climo_0.9x1.25_c20170724.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_num_bc_a4_anthro_surface_2000climo_0.9x1.25_c20170608.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_num_bc_a4_bb_surface_2000climo_0.9x1.25_c20170322.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_num_pom_a4_anthro_surface_2000climo_0.9x1.25_c20170608.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_num_pom_a4_bb_surface_2000climo_0.9x1.25_c20170509.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_num_so4_a1_anthro-ag-ship_surface_2000climo_0.9x1.25_c20170616.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_num_so4_a1_anthro-ene_vertical_2000climo_0.9x1.25_c20170616.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_num_so4_a1_bb_surface_2000climo_0.9x1.25_c20170322.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_num_so4_a2_anthro-res_surface_2000climo_0.9x1.25_c20170616.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_pom_a4_anthro_surface_2000climo_0.9x1.25_c20170608.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_pom_a4_bb_surface_2000climo_0.9x1.25_c20170322.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_SO2_anthro-ag-ship-res_surface_2000climo_0.9x1.25_c20170616.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_SO2_anthro-ene_surface_2000climo_0.9x1.25_c20170616.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_SO2_bb_surface_2000climo_0.9x1.25_c20170322.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_SO2_contvolcano_vertical_2000climo_0.9x1.25_c20170322.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_so4_a1_anthro-ag-ship_surface_2000climo_0.9x1.25_c20170616.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_so4_a1_anthro-ene_vertical_2000climo_0.9x1.25_c20170616.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_so4_a1_bb_surface_2000climo_0.9x1.25_c20170322.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_so4_a1_contvolcano_vertical_2000climo_0.9x1.25_c20170724.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_so4_a2_anthro-res_surface_2000climo_0.9x1.25_c20170616.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_so4_a2_contvolcano_vertical_2000climo_0.9x1.25_c20170724.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_SOAGx1.5_anthro_surface_2000climo_0.9x1.25_c20170608.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_SOAGx1.5_bb_surface_2000climo_0.9x1.25_c20170322.nc
atm/cam/chem/emis/CMIP6_emissions_2000climo/emissions-cmip6_SOAGx1.5_biogenic_surface_2000climo_0.9x1.25_c20170322.nc
atm/cam/chem/emis/elev/H2O_emission_CH4_oxidationx2_elev_3Dmonthly_L70_2000climo_c180511.nc
atm/cam/chem/trop_mozart/dvel/clim_soilw.nc
atm/cam/chem/trop_mozart/dvel/depvel_monthly.nc
atm/cam/chem/trop_mozart/dvel/regrid_vegetation.nc
atm/cam/chem/trop_mozart/dvel/season_wes.nc
atm/cam/chem/trop_mozart/phot/exo_coldens.nc
atm/cam/chem/trop_mozart/ub/clim_p_trop.nc
glc/cism/Greenland/glissade/init/greenland_4km_epsg3413_c171126.nc
lnd/clm2/mappingdata/maps/1.9x2.5/map_0.5x0.5_nomask_to_1.9x2.5_nomask_aave_da_c120709.nc
lnd/clm2/mappingdata/maps/1.9x2.5/map_1.9x2.5_nomask_to_0.5x0.5_nomask_aave_da_c120522.nc
```

Then we will have other sub-folders if the user choose another configuration.  I am not sure what is best for end-users.

Comment (bgruening) [reply tools/cesm/cesm.xml:148]: lets go with your mapping for a start... we can discuss maybe during the hackathon different solutions ....

Maybe repeats of repeats, that model the folder structures and people need to choose on any repeat level the files that should go into this "folder".

**Review by bgruening — APPROVED** [review]
Looks good from my side! :) Thanks Anne!
